trip_app/trips/views.py

```diff
@@ -1,23 +1,33 @@
 from core.utils import str_to_geopoint
 from django.contrib.gis.db.models.functions import Distance
 from django.db.models import F
-from rest_framework.generics import ListCreateAPIView
+from rest_framework import status
+from rest_framework.decorators import action
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
 
-from .models import Trip
-from .serializers import TripSerializer
+from .models import Trip, TripRequest
+from .permissions import IsTripDriverOrAdmin
+from .serializers import TripSerializer, TripRequestSerializer
 
 
-class TripListCreateApiView(ListCreateAPIView):
+class TripViewSet(ModelViewSet):
     serializer_class = TripSerializer
-    permission_classes = [IsAuthenticated, ]
     pagination_class = PageNumberPagination
     pagination_class.page_size = 5
 
     def perform_create(self, serializer):
         serializer.save(driver=self.request.user)
 
+    def get_permissions(self):
+        if self.action in ['update', 'destroy', 'partial_update']:
+            permission_classes = [IsTripDriverOrAdmin]
+        else:
+            permission_classes = [IsAuthenticated]
+        return [permission() for permission in permission_classes]
```

Comment:
This could be changed to: 
```
def get_permissions(self):
    if self.action in ['update', 'destroy', 'partial_update']:
            self.permission_classes = [IsTripDriverOrAdmin]
        else:
            self.permission_classes = [IsAuthenticated]
        return super().get_permissions()

```